src/promptflow-core/promptflow/executor/_tool_resolver.py

```diff
@@ -225,33 +230,63 @@ def _load_script_tool_as_function(self, node_name: str, assistant_tool: Assistan
         # Step IV: construct the AssistantTool object from the updated predefined_inputs + Tool object
         return self._construct_assistant_tool(tool, callable, updated_predefined_inputs)
 
-    def _generate_tool_definition(self, func_name: str, description: str, predefined_inputs: list) -> dict:
+    def _generate_tool_definition(self, tool: Tool, func: Callable, predefined_inputs: dict) -> dict:
```

Comment:
`_generate_assistant_tool_definition` to make it clear that it is the "tool" in assistant scenario not the "tool" in PF?